File: binary_search.exs

defmodule BS do

  def search(max) do
    min = 0
    target = random_from(min, max)
    IO.puts("#{target} is the target") 
    s(target, min, max, average(min, max))
  end

  def s( target, min, max, guess ) when target == guess do
    IO.puts(target)
  end 

  def s(target, min, max, guess ) do
    if target < guess do
      mx = guess - 1
      IO.puts("#{guess} too high") 
      s(target, min, mx, average(min, mx))
    else
      mn = guess + 1
      IO.puts("#{guess} too low") 
      s(target, mn, max, average(mn, max))
    end
  end 

  def average(min, max) do
    trunc ((max + min)/2)
  end

  def random_from(min, max) do
    trunc :random.uniform * (max - min) + min
  end
end

BS.search(300)

File: binarySearch.js
 //Picks the middle number, rounded down, within the range

function binarySearch(max) {
 var stat;
 var min = 0;
 var guessCount = 1;
 var guess = average(min, max); 
 var num = randomFromRange(min, max)

 function average(min, max) {
   return Math.floor((max + min) / 2 );
 }
 function randomFromRange(min, max) {
  return Math.floor(Math.random() * (max - min) + min);
 }

   while(guess !== num) {
     if(num < guess) {
       stat = 1;
       max = guess -1;
     }
     else {
       stat = 0;
       min = guess + 1;
     }
     
     print(message(guessCount, guess, stat));
     guess = average(min, max);
     guessCount++;
  } 
  stat = 2;
  print(message(guessCount, guess, stat));
}

binarySearch(3000);


function message(gn, g, s) { 
  var statuses = {
    0: "Too Low",
    1: "Too High",
    2: "You Found It!"
  }
  return "Guess " + gn + " Guessed: " + g + " Status: " + statuses[s]; 
}

function print(str) {
  console.log(str)
}
